app/models/task.rb

```diff
@@ -20,6 +20,7 @@
 
 class Task < ApplicationRecord
   has_and_belongs_to_many :campaigns
+  has_and_belongs_to_many :events
```

Comment:
I'm not a big fan of "has_and_belongs_to_many". I find that sooner or later we end up needing a model for the "join" table, so I prefer using has_many :through for most cases (for instance we might want to order the tasks in some special order inside a campaign or event). It might be over-engineering from where we are. But normally its worth it. Let me know what you think in this case!

There was an interesting article about this: http://blog.hasmanythrough.com/2006/4/20/many-to-many-dance-off, it's a bit old and you might have seen it already. But I always liked it. =) There's no clear winner and each works for its use case, so it's not really me pointing to "see this is the best!" =p It's just foi!